### PR TITLE
docs: Privacy settings manual scenarios

### LINKED ISSUE #21963: Privacy settings: Check “network details check” toggle works correctly

### PR DESCRIPTION
**Pull request by benjisclowder**
## **Description**

This PR documents the manual test scenarios for several privacy settings functionalities. Below is a summary of the validations included in these manual scenarios:

1. Privacy settings: Check “network details check” toggle works correctly

- Validate that the 'Network details check' toggle is present in the Security & Privacy settings and is ON by default.
- Validate that the toggle can be turned on and off.
- Validate that when the toggle is turned on, MetaMask checks the network details against the information provided by chainid.network when attempting to connect to a blockchain network.

2. Privacy settings: Check “IPFS gateway” toggle works correctly

- Validate that the 'IPFS gateway' toggle is present in the Security & Privacy settings and is ON by default.
- Validate that the toggle can be turned on and off.
- Validate that when the toggle is turned on, MetaMask shows images of NFTs.
- Validate that when the toggle is turned off, NFT images are replaced by placeholders and I am prompted to turn on IPFS resolution.

3. Privacy settings: Check “Show ENS domains in address bar” toggle works correctly

- Validate that the 'Show ENS domains in address bar' toggle is present in the Security & Privacy settings and is ON by default.
- Validate that the toggle can be turned on and off.
- Validate that when the toggle is turned on, ENS domain name is displayed alongside the public address.

4. Privacy settings: Check “Display NFT Media” toggle works correctly

- Validate that the 'Display NFT media' toggle is present in the Security & Privacy settings and is ON by default.
- Validate that the toggle can be turned on and off.
- Validate that when the toggle is turned on, MetaMask displays NFT media and data.
- Validate that NFT autodetection relies on this setting and won't be available when this is turned off.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26764?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/21963, https://github.com/MetaMask/metamask-extension/issues/21964, https://github.com/MetaMask/metamask-extension/issues/21966, https://github.com/MetaMask/metamask-extension/issues/21967

## **Manual testing steps**

Please click link for the [rendered view](https://github.com/MetaMask/metamask-extension/tree/privacy-settings-manual-scenarios/test/manual-scenarios/privacy-settings)

--> test/manual-scenarios/privacy-settings/check "Network details check" toggle works correctly.md
https://github.com/user-attachments/assets/9a60b564-7503-44b9-9ca0-810e857a4529

--> test/manual-scenarios/privacy-settings/check "IPFS gateway" toggle works correctly.md
https://github.com/user-attachments/assets/3e2e5135-9343-4d8c-808f-e3f599928d52

--> test/manual-scenarios/privacy-settings/check "Show ENS domains in address bar" toggle works correctly.md
https://github.com/user-attachments/assets/79df9cc4-0083-458f-be06-d524c7ca135f

--> test/manual-scenarios/privacy-settings/check "Display NFT media" toggle works correctly.md
https://github.com/user-attachments/assets/a54ef74f-00b4-4554-a7ac-9cd1e53800a4

### CODE REVIEW

File: test/manual-scenarios/privacy-settings/check "Network details check" toggle works correctly.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+When I click on the "Network details check" toggle icon
+Then the toggle switch should visually indicate to OFF
+When I click on the "Network details check" toggle icon again
+Then the toggle switch should visually indicate to ON
+
+# Scenario: Verification of network details check functionality
+
+Given I am on the Security & Privacy settings page
+And the "Network details check" toggle is set to ON
+When I attempt to connect to a blockchain network
```
Comment (hjetpoluru) [inline test/manual-scenarios/privacy-settings/check "Network details check" toggle works correctly.md:39]: I am unsure about the functionality of this toggle. Here is what I currently understand from the code: When the toggle is turned ON, the useSafeChainsListValidation flag is set to true in the state log. When this flag is true, if there are any errors, a request will be made to get the chain ID. My assumption is that when there is an error in fetching the chain ID in the network details, a request to retrieve the chain ID should be made.

Additionally, please refer to this ticket for more details and include the manual scenario mentioned in the ticket: [MetaMask Extension PR #20816](https://github.com/MetaMask/metamask-extension/pull/20816).

cc @seaona for more details

Comment (benjisclowder) [reply test/manual-scenarios/privacy-settings/check "Network details check" toggle works correctly.md:39]: @hjetpoluru This is the user-facing explanation of the toggle:
`Network details check
MetaMask uses a third-party service called chainid.network to show accurate and standardized network details. This reduces your chances of connecting to malicious or incorrect network. When using this feature, your IP address is exposed to chainid.network.`

In my opinion it's pretty straightforward but maybe I'm wrong?

Also these are the differences in the add network tab, first picture is with toggle ON, second with toggle OFF. Suggestions made on network details to ensure the user adds correct network data.

image (8)
image (9)

Comment (benjisclowder) [reply test/manual-scenarios/privacy-settings/check "Network details check" toggle works correctly.md:39]: Another example here: I entered network name, RPC URL and currency symbol for the Binance network, but the chainID is not correct so it suggests another name and currency symbol that matches the chainID, however it leaves out RPC URL for a reason I'm not aware of. The chainID flag saying `The RPC URL you have entered returned a different chain ID (56). Please update the Chain ID to match the RPC URL of the network you are trying to add.` is unrelated to toggle ON or OFF, it's present with any toggle status. 
image

Comment (hjetpoluru) [reply test/manual-scenarios/privacy-settings/check "Network details check" toggle works correctly.md:39]: @benjisclowder thanks, I understand the explanation. It gives me clarity on the toggle functionality you have provided. 

But I am still confused about what exactly the behavior should be at this point and what the necessary validation should be. If the screenshot you initially provided is accurate, then we should update the test scenario accordingly. Also, are there any other test scenarios where calling the chain ID site needs to be validated?

Comment (benjisclowder) [reply test/manual-scenarios/privacy-settings/check "Network details check" toggle works correctly.md]: Thanks a lot for the feedback, @hjetpoluru and @seaona! The updated scenario should now better reflect the toggle functionality from the user's expectations side. I would greatly appreciate one last review!

File: test/manual-scenarios/privacy-settings/check "Display NFT media" toggle works correctly.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+When I click on the "Display NFT media" toggle icon again
+Then the toggle switch should visually indicate to ON
+
+# Scenario: Verification of NFT media display functionality
+
+Given I am on the Security & Privacy settings page
+And the "Display NFT media" toggle is set to ON
+When I view my NFTs in MetaMask
+Then MetaMask should display NFT media and data
+And NFT autodetection should be available
```
Comment (hjetpoluru) [inline test/manual-scenarios/privacy-settings/check "Display NFT media" toggle works correctly.md:41]: @benjisclowder, we could remove this line after the discussion.

**Review by hjetpoluru — DISMISSED** [review]
LGTM

**Review by chloeYue — COMMENTED** [review]
Hi @benjisclowder , the scenarios looks good for me. Could you remove this change in the PR? 
Screenshot 2024-09-05 at 14 34 28

**Comment by hjetpoluru** [comment]
@chloeYue, good observation!! @benjisclowder, I removed the folder.

**Review by hjetpoluru — DISMISSED** [review]

**Review by chloeYue — APPROVED** [review]

**Review by hjetpoluru — APPROVED** [review]

**Pull request merged**